src/libraries/System.Text.RegularExpressions/gen/System.Text.RegularExpressions.Generator.csproj

```diff
@@ -18,7 +18,7 @@
 
   <ItemGroup>
     <!-- This generator ships as part of the .NET ref pack. Since VS ingests new Roslyn versions at a different cadence than the .NET SDK, we are pinning this generator to build against Roslyn 4.0 to ensure that we don't break users who are using .NET 7 previews in VS. -->
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion_4_0)" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion_4_X)" PrivateAssets="all" />
```

Comment:
Won't this mean that the regex source generator wouldn't work any longer in VS today? Correct me if I'm wrong but I believe VS doesn't yet have 4.2.x Roslyn on it, and because this generator is on by default and will run for any application, this would potentially just break creating new projects in VS. cc @ericstj

Reply:
@jaredpar @marcpopMSFT will know for sure.  I double checked VS2022 and after upgrading to 17.2 preview 2.1 I see it has a 4.2.0 build of Roslyn so it may be OK with latest.

Reply:
According to the release notes for 7.0.100 preview 2, the latest preview for VS2022 has a new enough Roslyn

Reply:
I wonder if we should be conservative just in case, especially when we don't yet need any new features from Roslyn. Main advantage that I see is that if folks are using an older VS2022 preview but with a new SDK, they would still be able to run.

Reply:
We need the 4.2.0 APIs for LibraryImportGenerator, so we need to pin that to this version at minimum when we put in in the ref pack anyway. Also, if we support running on an older VS preview with a newer SDK, then we should say that's the case in the download/release notes. If feel that requiring people who use preview builds to stay up to date isn't a bad thing, especially if our final release requirements will be even higher than these.

Reply:
If we really feel strongly about it, we can hold back the Regex generator, but I don't think it's worthwhile since we're planning on adding the LibraryImport generator soon too.

Reply:
```suggestion
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion_4_0)" PrivateAssets="all" />
```

Reply:
> We can hold off on RegexGenerator for now, but we will have this exact issue when LibraryImportGenerator is included in the ref pack (as it will be on by default as well), which is likely going to happen in a matter of weeks.

Right, I think the main issue here is, the moment we make this change (for the RegexGenerator for now only) then folks using older previews with new SDKs will be automatically broken, and at least as of today, RegexGenerator doesn't necessarily need this change yet. Making this change on the LibraryImportGenerator won't cause this yet, since as you pointed out, it is not yet part of the ref pack. Basically my thinking here, is that we should delay this break until we need it, and when we do, we should notify @marcpopMSFT so that they can make sure that the version we will depend on in the ref pack will be the one present in VS. At that point, we will have no choice but to break folks on older previews but we would need to do so because we depended on the new compiler so I think it would be better justified.